Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.TypeBinaryExpression.cs

```diff
@@ -52,5 +52,27 @@ public Type TypeOperand
         return default(Type);
       }
     }
+
+#if NETFRAMEWORK_4_0 || SILVERLIGHT_4_0 || SILVERLIGHT_5_0
+    //
+    // Summary:
+    //     Creates a new expression that is like this one, but using the supplied children.
+    //     If all of the children are the same, it will return this expression.
+    //
+    // Parameters:
+    //   expression:
+    //     The System.Linq.Expressions.TypeBinaryExpression.Expression property of the result.
+    //
+    // Returns:
+    //     This expression if no children are changed or an expression with the updated
+    //     children.
+    [Pure]
+    public TypeBinaryExpression Update(Expression expression)
+    {
+      Contract.Requires(expression != null);
+      Contract.Ensures(Contract.Result<TypeBinaryExpression>() != null);
```

Comment:
From this: http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/TypeBinaryExpression.cs,a071b1eaa5acb0f6

`Contract.Requires(TypeOperand != null && !TypeOperand.IsByRef);`